### PR TITLE
Update GH Actions to align to new guidelines

### PR DESCRIPTION
**Pull request by QcFe**
# Description

This PR changes the dramatic `set-output::stuff` to the new `>> $GITHUB_OUTPUT`, in order to remove deprecation warnings and align to new GH Actions format. This also updates `retrieve-build-matrix` custom action to latest version of `@actions/core`.

### CODE REVIEW

**Comment by frisso** [comment]
Many many thanks, @QcFe !

**Comment by giorio94** [comment]
@QcFe Since you are playing with GH Actions, it might be worth to bump a few actions, to remove also the warnings about the deprecated node version (and probably also the remaining ones about save-state)...

File: provisioning/containers/gui-common/tigervnc/Xvnc-session [hunk 1]
```diff
@@ -5,8 +5,8 @@
 # > add xhost (to enable connections to the X server from external containers)
 # > start window manager (fluxbox)
 
-test x"$SHELL" = x"" && SHELL=/bin/bash
-test x"$1"     = x"" && set -- default
+test "$SHELL" = x"" && SHELL=/bin/bash
+test "$1"     = x"" && set -- default
```
Comment (giorio94) [inline provisioning/containers/gui-common/tigervnc/Xvnc-session:9]: This doesn't seem intended to me.

**Review by giorio94 — APPROVED** [review]

**Pull request merged**